core/trino-main/src/main/java/io/trino/operator/aggregation/AccumulatorCompiler.java

```diff
@@ -678,8 +676,8 @@ private static void generateAddIntermediateAsCombine(
         loopBody.comment("combine(state_0, state_1, ... scratchState_0, scratchState_1, ... lambda_0, lambda_1, ...)");
         for (FieldDefinition stateField : stateFields) {
             if (grouped) {
-                Variable groupIdsBlock = scope.getVariable("groupIdsBlock");
-                loopBody.append(thisVariable.getField(stateField).invoke("setGroupId", void.class, groupIdsBlock.invoke("getGroupId", long.class, position)));
+                Variable groupIds = scope.getVariable("groupIds");
+                loopBody.append(thisVariable.getField(stateField).invoke("setGroupId", void.class, groupIds.getElement(position).cast(long.class)));
```

Comment:
Does it make sense to also make `io.trino.spi.function.GroupedAccumulatorState#setGroupId` accept `int` instead of `long`?

Reply:
Yes!  That will be in some follow up work.